test/src/tests-session-manager.ts

```diff
@@ -0,0 +1,607 @@
+                },
+            };
+
+            mParticle.init(apiKey, window.mParticle.config);
+
+            const mpInstance = mParticle.getInstance();
+            const callbackSpy = sinon.spy();
+            const identifySpy = sinon.spy(mpInstance.Identity, 'identify');
+
+            mpInstance._Store.SDKConfig.identifyRequest = identityApiData;
+            mpInstance._Store.SDKConfig.identityCallback = callbackSpy;
+            mpInstance._Store.identifyCalled = false;
+            //
+            mpInstance._SessionManager.startNewSession();
+
+            expect(identifySpy.getCall(0).args[0]).to.eql(identityApiData);
+            expect(identifySpy.getCall(0).args[1]).to.eql(callbackSpy);
+            expect(callbackSpy.called).to.equal(true);
+            expect(mpInstance._Store.identifyCalled).to.eql(true);
+            expect(mpInstance._Store.SDKConfig.identityCallback).to.eql(null);
+        });
+    });
+
+    describe('#endSession', () => {
+        // TODO: Set up scaffolding for session
+        it.skip('should end a session', () => {
+            mParticle.init(apiKey, window.mParticle.config);
+            const mpInstance = mParticle.getInstance();
+            const persistenceSpy = sinon.spy(mpInstance._Persistence, 'update');
+
+            mpInstance._SessionManager.endSession();
+
+            expect(mpInstance._Store.sessionId).to.equal(null);
+            expect(mpInstance._Store.dateLastEventSent).to.equal(null);
+            expect(mpInstance._Store.sessionAttributes).to.eql({});
+
+            // Persistence isn't necessary for this feature, but we should test
+            // to see that it is called in case this ever needs to be refactored
+            expect(persistenceSpy.called).to.equal(true);
+        });
+
+        it('should force a session end when override is used', () => {
+            mParticle.init(apiKey, window.mParticle.config);
+            const mpInstance = mParticle.getInstance();
+            const persistenceSpy = sinon.spy(mpInstance._Persistence, 'update');
+
+            mpInstance._SessionManager.endSession(true);
+
+            expect(mpInstance._Store.sessionId).to.equal(null);
+            expect(mpInstance._Store.dateLastEventSent).to.equal(null);
+            expect(mpInstance._Store.sessionAttributes).to.eql({});
+
+            // Persistence isn't necessary for this feature, but we should test
+            // to see that it is called in case this ever needs to be refactored
+            expect(persistenceSpy.called).to.equal(true);
+        });
+
+        it('should do nothing when a session does not exist', () => {});
+        it('should do nothing when persistence cookies are missing', () => {});
+        it('should nullify session attributes when override is used', () => {});
+        it('should use the cookie sessionId over Store.sessionId', () => {});
+
+        it('should NOT end session if session has not timed out', () => {
```

Comment:
Perhaps add a comment that 20 minutes ago is less than the default session time out of 30.